### PR TITLE
PR for STACK-264

### PR DESCRIPTION
**Pull request by hthompson6**

### CODE REVIEW

File: tests/unit/client/test_axapi_http.py [hunk 1]
```diff
@@ -11,3 +11,101 @@
...
+import requests
+import socket
+import unittest
+
+from a10_saltstack.client.axapi_http import HttpClient 
+
+
+class TestAxapiHTTPClient(unittest.TestCase):
+
+    def _patch_wrap(self, path):
```
Comment (mdurrant-b3) [inline tests/unit/client/test_axapi_http.py]: Excessive patching is a code smell.  Use responses to test requests.

**Review by ghost — CHANGES_REQUESTED** [review]
Can we look at using `responses` for the HTTP lib testing? 

If not possible, please respond to this comment with the reasoning.

**Comment by hthompson6** [comment]
> Can we look at using `responses` for the HTTP lib testing?
> 
> If not possible, please respond to this comment with the reasoning.

Make no mistake. The responses library is itself just a patch of the requests library that's been wrapped with extra functionality. It's biggest contribution is the inclusion of callbacks and ensuring the that desired URL is hit by default.

I see why this library could be useful, but we don't have enough tests in which we care about the response, the url, and the method type to warrant using it. It just wouldn't get us anything extra in terms of testing or readability at least so far as how we'd end up using it.

Instead, I've gone about removing some dependencies and imports that weren't be using. I have noticed that the Session class has been utilized differently in the acos client version of the AXAPI to allow for max retries and timeouts. I'll add a story to investigate and mirror those changes.

**Comment by mdurrant-b3** [comment]
👍

**Review by ghost — APPROVED** [review]

**Pull request merged**